homeassistant/components/nightscout/sensor.py

```diff
@@ -28,10 +30,24 @@ async def async_setup_entry(
     entry: ConfigEntry,
     async_add_entities: AddEntitiesCallback,
 ) -> None:
-    """Set up the Glucose Sensor."""
-    api = hass.data[DOMAIN][entry.entry_id]
+    """Set up entities."""
+    data = hass.data[DOMAIN][entry.entry_id]
+    api = data[API]
+    coordinator = data[COORDINATOR]
+
+    # Glucose sensor
     async_add_entities([NightscoutSensor(api, "Blood Sugar", entry.unique_id)], True)
 
+    # Uploader batteries
+    entities = []
+
+    for device_name in coordinator.data:
+        entities.append(
+            Battery(coordinator, device_name, f"{entry.unique_id}_{device_name}")
```

Comment:
Is the device name both unique and stable? Can the user change the device name?

Reply:
I think if the user changes, for example, the phone name, it will be instantiated as a new sensor. But the Nightscout API doesn't have any other IDs to relay on, just the name.

Reply:
We don't allow non stable ids as unique id.

https://developers.home-assistant.io/docs/entity_registry_index#unique-id-requirements